### PR TITLE
ENH: Fix issues found by lgtm tool

### PR DESCRIPTION
**Pull request by wagner-certat**
Most notably: Implemented `Message.__eq__` for proper comparisons

See: https://lgtm.com/projects/g/certtools/intelmq
Many of them are false positives, but some hints are very good :)

### CODE REVIEW

File: intelmq/lib/message.py [hunk 1]
```diff
@@ -443,6 +445,25 @@ def to_json(self, hierarchical=False, with_type=False, jsondict_as_string=False)
...
+            type_eq = type(self) == type(other)
+            harm_eq = self.harmonization_config == other.harmonization_config if hasattr(other, 'harmonization_config') else False
+            if type_eq and dict_eq and harm_eq:
+                return True
+        elif dict_eq:
+            return True
+
+    def __ne__(self, other) -> bool:
+        return not self.__eq__(other)
+
```
Comment (SYNchroACK) [inline intelmq/lib/message.py]: I think its missing a final `return False`.

File: intelmq/lib/message.py [hunk 2]
```diff
@@ -443,6 +445,25 @@ def to_json(self, hierarchical=False, with_type=False, jsondict_as_string=False)
...
+        """
+        dict_eq = super(Message, self).__eq__(other)
+        if issubclass(type(other), Message):
+            type_eq = type(self) == type(other)
+            harm_eq = self.harmonization_config == other.harmonization_config if hasattr(other, 'harmonization_config') else False
+            if type_eq and dict_eq and harm_eq:
+                return True
+        elif dict_eq:
+            return True
+
```
Comment (SYNchroACK) [inline intelmq/lib/message.py]: I think its missing a final `return False`.

Comment (wagner-certat) [reply intelmq/lib/message.py]: Implicitly it's None here.
According to the docs, if the return value is not boolean, bool() is called.

Pushed an additional commit

Comment (SYNchroACK) [reply intelmq/lib/message.py]: Yes, its true but I tested here and it gaves me `None`, not `False`. May be I did something wrong.

```
>>> def return_test(other) -> bool:
...   if other == "a":
...     return True
... 
>>> print( return_test("a") )
True
>>> print( return_test("b") )
None
```

**Pull request merged**